plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart

```diff
@@ -0,0 +1,262 @@
+        log.add(methodCall);
+        return expectedItemsForNamespace1;
+      });
+    });
+
+    test('is validated', () async {
+      //test with getNamespaces()
+      final int actualGetNameSpacesLength = idMap.getNamespaces().length;
+      expect(actualGetNameSpacesLength, equals(2));
+
+      final List<IdentityItem> actualItemsForNamespace1 =
+          idMap.getIdentityItemsForNamespace('namespace1');
+
+      //test identityItems in namespace, getIdentityItemsForNamespace()
+      final int actualItemsForNamespace1Length =
+          idMap.getIdentityItemsForNamespace('namespace1').length;
+
+      expect(actualItemsForNamespace1Length, equals(2));
+
+      expect(actualItemsForNamespace1[0].id, expectedItemsForNamespace1[0].id);
+      expect(actualItemsForNamespace1[0].authenticatedState,
+          expectedItemsForNamespace1[0].authenticatedState);
+      expect(actualItemsForNamespace1[0].primary,
+          expectedItemsForNamespace1[0].primary);
+      expect(actualItemsForNamespace1[1].id, expectedItemsForNamespace1[1].id);
+      expect(actualItemsForNamespace1[1].authenticatedState,
+          expectedItemsForNamespace1[1].authenticatedState);
+      expect(actualItemsForNamespace1[1].primary,
+          expectedItemsForNamespace1[1].primary);
+    });
+  });
+
+  group('IdentityMap getIdentityItemsForNamespace', () {
+    IdentityItem item1 =
+        new IdentityItem('id1', AuthenticatedState.AMBIGUOUS, true);
+    IdentityItem item2 =
+        new IdentityItem('id2', AuthenticatedState.AUTHENTICATED, false);
+    IdentityItem item3 = new IdentityItem('id3', AuthenticatedState.LOGGED_OUT);
+    IdentityItem item4 = new IdentityItem('id4');
+
+    IdentityMap idMap = new IdentityMap();
+    idMap.addItem(item1, 'namespace1');
+    idMap.addItem(item2, 'namespace1');
+    idMap.addItem(item3, 'namespace2');
+    idMap.addItem(item4, 'namespace2');
+
+    List<IdentityItem> expectedItemsForNamespace1 = <IdentityItem>[
+      item1,
+      item2
+    ];
+
+    final List<MethodCall> log = <MethodCall>[];
+
+    setUp(() {
+      channel.setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+        return expectedItemsForNamespace1;
+      });
+    });
+
+    test('test a non-existed namespace', () async {
+      final int actualItemsForNamespace3Length =
+          idMap.getIdentityItemsForNamespace('namespace3').length;
+      expect(actualItemsForNamespace3Length, equals(0));
+    });
+
+    test('test an invalid namespace', () async {
+      final int actualItemsForNamespaceInvalidLength =
+          idMap.getIdentityItemsForNamespace('invalid').length;
+      expect(actualItemsForNamespaceInvalidLength, equals(0));
+    });
+
+    test('test an empty namespace', () async {
+      final int actualItemsForNamespaceInvalidLength =
+          idMap.getIdentityItemsForNamespace('invalid').length;
+      expect(actualItemsForNamespaceInvalidLength, equals(0));
```

Comment:
testing empty string here instead of invalid? 
actualItemsForNamespaceEmptyLength